azbfs/datalake_storage_transforms.go

```diff
@@ -0,0 +1,31 @@
+package azbfs
+
+import (
+	"encoding/base64"
+	"strings"
+)
+
+// Converts metadata into a string of format "key1=value1, key2=value2" and Base64 encodes the values.
+func BuildMetadataString(md map[string]string) *string {
+	if len(md) == 0 {
```

Comment:
for reference for other reviewers: maps are nullable, and when nil, maps and lists return 0 on len. [This is safe.](https://play.golang.org/p/XaFWmYwgVCB)